src/helpers/models.rs

```diff
@@ -0,0 +1,143 @@
+use serde::{Deserialize, Serialize};
+use std::fmt::{Debug, Formatter};
+use std::ops::Range;
+
+// Type aliases to indicate whether the parameter should be encrypted, secret shared, etc.
+// Underlying types are temporalily assigned for PoC.
+type CipherText = [u8; 32];
+type PlainText = String;
+type SecretShare = [CipherText; 3];
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+struct Event {
+    /// Name of the business entity that generated this event. For source-fanout,
+    /// it's the business who promotes ads. For trigger-fanout, it's the business
+    /// where a user has made a conversion.
+    entity_name: PlainText,
+
+    /// Secret shared and then encrypted match keys.
+    matchkey: SecretShare,
+
+    /// Date and time of the event occurence. Secret shared and encrypted.
+    timestamp: SecretShare,
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+struct SourceEvent {
+    event: Event,
+
+    /// Campaign ID of the ad served to the user.
```

Comment:
Per this thread: https://github.com/patcg/private-measurement/issues/9#issuecomment-1121329028 we might want to consider renaming this to "group_id" or even just "breakdown_key".